SU2_CFD/src/fluid/CFluidFlamelet.cpp

```diff
@@ -0,0 +1,150 @@
+    table_lookup_names[i_lookup] = config->GetLUTLookupName(i_lookup);
+  }
+
+  Pressure = value_pressure_operating;
+
+  PreprocessLookUp();
+}
+
+CFluidFlamelet::~CFluidFlamelet() {
+  delete look_up_table;
+}
+
+void CFluidFlamelet::SetTDState_T(su2double val_temperature, const su2double* val_scalars) {
+  su2double val_enth = val_scalars[I_ENTH];
+  su2double val_prog = val_scalars[I_PROGVAR];
+
+  /*--- Add all quantities and their names to the look up vectors. ---*/
+  look_up_table->LookUp_XY(varnames_TD, val_vars_TD, val_prog, val_enth);
+
+  Temperature = val_vars_TD[GetIdx(varnames_TD,"Temperature")];
+  Density = val_vars_TD[GetIdx(varnames_TD,"Density")];
+  Cp = val_vars_TD[GetIdx(varnames_TD,"Cp")];
+  Mu = val_vars_TD[GetIdx(varnames_TD,"ViscosityDyn")];
+  Kt = val_vars_TD[GetIdx(varnames_TD,"Conductivity")];
+  mass_diffusivity = val_vars_TD[GetIdx(varnames_TD,"DiffusionCoefficient")];
+  molar_weight = val_vars_TD[GetIdx(varnames_TD,"MolarWeightMix")];
+
+  /*--- Compute Cv from Cp and molar weight of the mixture (ideal gas). ---*/
+  Cv = Cp - UNIVERSAL_GAS_CONSTANT / molar_weight;
+}
+
+/* --- Total enthalpy is the transported variable, but we usually have temperature as a boundary condition,
+       so we do a reverse lookup */
+unsigned long CFluidFlamelet::GetEnthFromTemp(su2double& val_enth, const su2double val_prog, const su2double val_temp,
+                                              const su2double initial_value) {
+
+  /*--- convergence criterion for temperature in [K], high accuracy needed for restarts. ---*/
+  su2double delta_temp_final = 0.001;
+  su2double enth_iter = initial_value;
+  su2double delta_enth;
+  su2double delta_temp_iter = 1e10;
+  unsigned long exit_code = 0;
+  const int counter_limit = 1000;
+
+  int counter = 0;
+  while ((abs(delta_temp_iter) > delta_temp_final) && (counter++ < counter_limit)) {
+    /*--- Add all quantities and their names to the look up vectors. ---*/
+    look_up_table->LookUp_XY(varnames_TD, val_vars_TD, val_prog, enth_iter);
+    Temperature = val_vars_TD[GetIdx(varnames_TD,"Temperature")];
+    Cp = val_vars_TD[GetIdx(varnames_TD,"Cp")];
+
+    delta_temp_iter = val_temp - Temperature;
+
+    delta_enth = Cp * delta_temp_iter;
+
+    enth_iter += delta_enth;
+  }
+
+  val_enth = enth_iter;
+
+  if (counter >= counter_limit) {
+    exit_code = 1;
+  }
+
+  return exit_code;
+}
+
+void CFluidFlamelet::PreprocessLookUp() {
+  /*--- Set lookup names for all relevant lookup processes in the fluid model. ---*/
+
+  /*--- Thermodynamic state variables and names. ---*/
+  varnames_TD.resize(7);
+  val_vars_TD.resize(7);
+
+  /*--- The string in varnames_TD as it appears in the LUT file. ---*/
+  varnames_TD[0] = "Temperature";
+  varnames_TD[1] = "Density";
+  varnames_TD[2] = "Cp";
+  varnames_TD[3] = "ViscosityDyn";
+  varnames_TD[4] = "Conductivity";
+  varnames_TD[5] = "DiffusionCoefficient";
+  varnames_TD[6] = "MolarWeightMix";
```

Comment:
I think you could just move this code to the constructor

Reply:
yes, I will remove preprocesslookup.